### PR TITLE
Add an --orderBy parameters to call-api

### PR DESCRIPTION
**Pull request by caberos**
https://github.com/softlayer/softlayer-python/issues/1453

### CODE REVIEW

File: SoftLayer/CLI/call_api.py [hunk 1]
```diff
@@ -112,14 +112,17 @@ def _validate_parameters(ctx, param, value):  # pylint: disable=unused-argument
 @click.option('--mask', help="String-based object mask")
 @click.option('--limit', type=click.INT, help="Result limit")
 @click.option('--offset', type=click.INT, help="Result offset")
+@click.option('--orderBy', type=click.STRING, help="an object filter that adds an order by clause"
```
[the remark this answers is not in the document]
Comment (ATGE) [reply SoftLayer/CLI/call_api.py:115]: or better add the examples between 149-150 lines, I mean within the doc of  `def cli `method

**Review by ATGE — COMMENTED** [review]
### Testing

- ✔️ `slcli -v call-api Account getVirtualGuests  --mask=id,hostname   --orderBy 'virtualGuests.id=ASC' --limit=5`

- ✔️slcli -v call-api Account getVirtualGuests  --mask=id,hostname   --orderBy 'virtualGuests.id'

- ❌` slcli  -v call-api Account getVirtualGuests  --mask=typeId,maxCpu   -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu=ASC' `

- ❌`slcli -v call-api Account getVirtualGuests  --mask=typeId,maxCpu   -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu'`
 also,~I don't~ I'm not sure if this feature should be supported orderBy as multiple params because a sorting query can contain many orderBy options e.g `slcli call-api --orderBy property1 --orderBy property2`

**Comment by allmightyspiff** [comment]
Both of these commands seem to work for me.  @ATGE do these still fail your testing, and if so can you add some detail on what went wrong?

✔️ `slcli -v call-api Account getVirtualGuests --mask=typeId,maxCpu -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu=ASC'`

✔️ `slcli -v call-api Account getVirtualGuests --mask=typeId,maxCpu -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu'`


```
$ ./slcli --format=table -v call-api Account getVirtualGuests --mask=typeId,maxCpu -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu=ASC' --limit=10
Calling: SoftLayer_Account::getVirtualGuests(id=None, mask='typeId,maxCpu', filter='{'virtualGuests': {'maxCpu': {'operation': 'orderBy', 'options': [{'name': 'sort', 'value': ['ASC']}]}}}', args=(), limit=10, offset=None))
:........:........:
: maxCpu : typeId :
:........:........:
:   1    :   1    :
:   1    :   1    :
:   1    :   1    :
:   1    :   1    :
:   1    :   4    :
:   1    :   1    :
:   2    :   1    :
:   2    :   1    :
:   2    :   1    :
:   2    :   1    :
:........:........:

$ ./slcli -v  --format=table call-api Account getVirtualGuests --mask=typeId,maxCpu -f 'virtualGuests.typeId=1' --orderBy 'virtualGuests.maxCpu'
Calling: SoftLayer_Account::getVirtualGuests(id=None, mask='typeId,maxCpu', filter='{'virtualGuests': {'maxCpu': {'operation': 'orderBy', 'options': [{'name': 'sort', 'value': ['DESC']}]}}}', args=(), limit=None, offset=None))
:........:........:
: maxCpu : typeId :
:........:........:
:   30   :   1    :
:   30   :   1    :
:   16   :   1    :
:   8    :   1    :
:   4    :   1    :
:   4    :   1    :
:   2    :   1    :
:   2    :   1    :
:   2    :   1    :

```

>also, I don't sure if this feature should be supported orderBy as multiple params because a sorting query can contain many orderBy options e.g slcli call-api --orderBy property1 --orderBy property2

For now, I think it is fine to just have a single sortBy for a CLI operation. If a user wants a more complex sorting, they will likely need to write their own API call / Python code for that.

❌ `slcli -v  --format=table call-api Account getVirtualGuests --mask=typeId,maxCpu,hostname --orderBy 'virtualGuests.maxCpu' --json-filter='{"virtualGuests":{"hostname":{"operation":"hourlydc"}}}'`

Using `json-filter` and `--orderBy` makes the slcli ignore the `orderBy` option.

```python
$ ./slcli -v  --format=table call-api Account getVirtualGuests --mask=typeId,maxCpu,hostname --orderBy 'virtualGuests.maxCpu' --json-filter='{"virtualGuests":{"hostname":{"operation":"hourlydc"}}}'
Calling: SoftLayer_Account::getVirtualGuests(id=None, mask='typeId,maxCpu,hostname', filter='{'virtualGuests': {'hostname': {'operation': 'hourlydc'}}}', args=(), limit=None, offset=None))
:..........:........:........:
: hostname : maxCpu : typeId :
:..........:........:........:
: hourlydc :   2    :   1    :
:..........:........:........:
```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Need a way to better merge object filters that share the same top level keys.
https://github.com/softlayer/softlayer-python/issues/1461 will create a utility to do that, then you can just update this code and it will be good to go.

Thanks

**Comment by allmightyspiff** [comment]
@caberos in #1468 I added a utility function `result = SoftLayer.utils.dict_merge(filter1, filter2)` that should let you easily merge object filters together for this issue. Once its in master you should be able to pull that change in and use it.

I thought about your suggestion of using `jsonmerge` but I felt adding that library would be a bit too much changes for our simple use case. I think this function should work well enough, but if you run into issues with it let me know.

**Review by allmightyspiff — APPROVED** [review]
looks good.

Only thing, please merge in the latest changes from master to get the build to pass properly. Thanks.

**Review by ATGE — APPROVED** [review]

**Review by FernandoOjeda — APPROVED** [review]
Looks good.

**Pull request merged**